lncfg/config.go

```diff
@@ -40,6 +40,10 @@ const (
 	// push us in the broadcast window.
 	DefaultFinalCltvRejectDelta = DefaultIncomingBroadcastDelta + 3
 
+	// DefaultAutoFailDelta is the number of blocks before expiry of an htlc
+	// when we'll automatically cancel back.
+	DefaultAutoFailDelta = 3
```

Comment:
Hmm yes, that's a good point. There is already a cut-off present in the interceptor after which a (new) htlc isn't offered to the interceptor at all anymore. We could use that same value for auto-fail, but then the situation may occur where the interceptor basically has no time to respond. So it'll probably need to be that cut-off value plus a few blocks?

As for other implementations, I don't think they'll go to chain early because they can only claim back the htlc after the expiry anyway?

Reply:
Changed this into `CltvInterceptDelta` and based it on `DefaultFinalCltvRejectDelta`.

Reply:
I'm a bit unsure about the logic in `handleExpired` and `failExpiredHtlcs` - I'm not sure that they're useful atm.

- In `handleExpired`, the default of 13 blocks is used for htlc's just coming in
- `failExpiredHtlcs` uses the default of 10 blocks for held htlc's

The minimum time lock delta that lnd allows is 18 blocks. So in the first case (similar thing applies for second case), if an htlc comes in at 14 blocks and goes through without a hitch, it will get failed when the switch calls `CheckHtlcForward` since the outgoing timeout MUST have passed. So I think these values still need to be larger to account for the time-lock-delta. It would ideally also account for `OutgoingCltvRejectDelta` (default: 3) which is used in the link (`CheckHtlcForward`->`canSendHtlc`).

Perhaps the problem is that we're comparing against the incoming expiry rather than the outgoing

Reply:
I see what you mean. I thought that it would be enough to just guard against force-closes in the interceptable switch. To not let it become too dependent on specific forwarding logic. 

The worst that can happen I think is that the interceptor client signals resume, and then the switch drops the packet anyway because of a forwarding or outgoing cltv delta violation.

Another thing is that in the case the interceptor doesn't resume but always either settles or fails directly, these two deltas are probably not relevant. For example in the case of forwarding over a virtual hop that doesn't actually exists.

Reply:
The reason that these `handleExpired` and `failExpiredHtlcs` exist is just to protect the channel against interceptor clients that become unresponsive. This isn't something that can be handled by the switch.

Reply:
In that case, I think `failExpiredHtlcs` should use 11 blocks instead of 10. The reason being that the contractcourt could act first and close the channel before `failExpiredHtlcs` runs. It is getting complicated to keep track of all of these constants though...

Reply:
I think it is currently configured at 13/16 blocks. So we only offer to the interceptor if more than 16 blocks left. So the interceptor has at least 3 blocks to respond. If it doesn't respond, we fail off-chain automatically and then there are still 3 blocks left until cnct wakes up. If I am not mistaking here...

Reply:
Yes you are right